.github/workflows/e2e_test.yml

```diff
@@ -3,32 +3,49 @@ name: E2E Tests
 on:
   workflow_dispatch:
   pull_request:
-  push:
-    branches:
-    - main
+    types:
+      # trigger workflow if PR is opened directly as R4R.
+      - opened
+      # trigger workflow if changes are pushed to the branch.
+      - synchronize
+      # trigger workflow if PR is marked ready for review.
+      - ready_for_review
+    paths-ignore:
+      - 'docs/**'
+      - '**.md'
+      - 'LICENSE'
 
 permissions:
   contents: read
   packages: write
 
-jobs:
-  test-ibc-transfer:
+jobs:   
+  build-image:
     runs-on: ubuntu-latest
     steps:
-        - name: Check out the repo
-          uses: actions/checkout@v3  
-        
-        - name: Set up Docker Buildx
-          uses: docker/setup-buildx-action@v3
-      
-        - name: Build image
-          id: build_image
-          run: docker build . -t "ghcr.io/dymensionxyz/dymension:e2e" 
+      - name: Checkout
+        uses: actions/checkout@v4
 
-        - name: Set up Go 1.19
-          uses: actions/setup-go@v3
-          with:
-            go-version: 1.19
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
 
-        - run: make e2e-test-ibc
-        - run: make e2e-test-ibc-timeout
\ No newline at end of file
+      - name: Build and export
+        uses: docker/build-push-action@v5
+        with:
+          file: Dockerfile
+          context: .
+          outputs: type=docker,dest=/tmp/e2e.tar
+          platforms: linux/amd64
+          tags: |
+              ghcr.io/dymensionxyz/dymension:e2e
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: e2e
+          path: /tmp/e2e.tar  
+  e2e-tests:
+    needs: build-image
+    uses: dymensionxyz/e2e-tests/.github/workflows/e2e-test-workflow-call.yml@87611e27234bc9b4b7d191f76a546049fa87d39f
```

Comment:
can you explain that?
* I don't see `e2e-test-workflow.yml` anywhere
* What is `@87611e27234bc9b4b7d191f76a546049fa87d39f` ?

Reply:
I have written a reusable workflow here: https://github.com/dymensionxyz/e2e-tests/tree/ducnt/move_tests/.github/workflows

Reply:
The hash code in this code is the commit hash of the reusable workflow

Reply:
And we can trigger the reusable workflow from different repositories.

Reply:
I get it but once we merge the current PR to default main branch, then your commit hash shouldn't be needed. Can you remove it?

Reply:
It has been removed!!